File: java/java/DequeProgram.java
import java.util.*;
public class DequeProgram{
    public static void main(String[] args){
      Deque<Integer> dq= new ArrayDeque<Integer>();
      dq.addFirst(10);
      dq.addFirst(20);
      dq.addLast(30);
      dq.addFirst(40);
      dq.addLast(50);
      System.out.println(dq);
      System.out.println(dq.getFirst());
      System.out.println(dq.getLast());
      System.out.println(dq.peekFirst());
      System.out.println(dq.peekLast());
      while(!dq.isEmpty()){
          System.out.println(dq.removeFirst());
      }
    }
}

File: kt/kotlin/DequeProgram.kt
import 	java.util.*
class LiveList(var x1:Int, var x2:Int)
fun main() {
    var dq = ArrayDeque<Int>()
    dq.addLast(4)
    dq.addFirst(10)
    dq.addLast(3)
    dq.addLast(12)
    dq.addFirst(2)
    println(dq)
    dq.removeFirst()
    println(dq)
}